asmcomp/arm64/emit.mlp

```diff
@@ -755,8 +752,6 @@ let emit_instr env i =
         emit_stack_adjustment (-n);
         env.stack_offset <- env.stack_offset + n
     | Lop(Iload { memory_chunk; addressing_mode; is_atomic }) ->
-        if is_atomic then ` dmb ishld\n`;
-        let load_type = if is_atomic then "ldar" else "ldr" in
         let dst = i.res.(0) in
```

Comment:
The code (I think) is assuming that `memory_chunk = Word_int || memory_chunk = Word_val || is_atomic = false`. Presumably there is no way (currently) that we end up with an atomic load of a `Double`.

```suggestion
        assert(memory_chunk = Word_int || memory_chunk = Word_val || is_atomic = false);
        let dst = i.res.(0) in
```

Reply:
OK for the assertion.  Atomic loads are generated only for the translation of `Atomic.get`, so only at `Word` type.